src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,189 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            linkFirst(value);
+        } else {
+            linkLast(value);
+        }
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("Out of bounds search index");
+        }
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        T[] addArray = (T[]) list.toArray();
+        int lengthOfArray = addArray.length;
+        if (lengthOfArray == 0) {
+            return false;
+        }
+        Node<T> prev;
+        prev = last;
+        for (T elementAddArray : addArray) {
+            Node<T> inputNode = new Node<>(elementAddArray, null, prev);
+            if (prev == null) {
+                first = inputNode;
+            } else {
+                prev.next = inputNode;
+            }
+            prev = inputNode;
+        }
+        size += lengthOfArray;
+        return true;
```

Comment:
can you just make use of add() method here?

Reply:
Of course. Done)
Why is addAll not implemented through add in the original class?